plugins/auth-node/src/oauth/createOAuthRouteHandlers.ts

```diff
@@ -248,11 +249,20 @@ export function createOAuthRouteHandlers<TProfile>(
         const { name, message } = isError(error)
           ? error
           : new Error('Encountered invalid error'); // Being a bit safe and not forwarding the bad value
-        // post error message back to popup if failure
-        sendWebMessageResponse(res, origin, {
-          type: 'authorization_response',
-          error: { name, message },
-        });
+
+        if (state?.flow === 'redirect' && state?.redirectUrl) {
+          const redirectUrl = new URL(state.redirectUrl);
+          redirectUrl.searchParams.set('error', encodeURIComponent(message));
```

Comment:
`searchParams` already encodes the value, so I think you're double encoding here and double decoding in the frontend